client/src/worldgen.rs

```diff
@@ -257,19 +264,59 @@ impl ChunkParams {
         }
     }
 
-    fn generate_road_support(&self, center: na::Vector3<f64>) -> Option<Material> {
+    fn generate_road_support(
+        &self,
+        center: na::Vector3<f64>,
+        coords: na::Vector3<u8>,
+    ) -> Option<Material> {
         let plane = -Plane::from(Side::B);
         let horizontal_distance = plane.elevation(center, self.chunk);
 
-        if horizontal_distance < 0.3 {
+        if (horizontal_distance < 0.3) && self.trussing_at(coords) {
```

Comment:
```suggestion
        if horizontal_distance < 0.3 && self.trussing_at(coords) {
```
Comparisons bind tighter than boolean operations.